lib/runtime/adapter-read.js

```diff
@@ -69,6 +69,11 @@ class AdapterRead {
     // Common read options that should be supported by all adapters.
     static get commonOptions() { return ['from', 'to', 'last', 'timeField', 'every', 'lag']; }
 
+    // subclasses can override this method to return an array
+    // if they do we'll check that all options passed to the
+    // proc are in the array
+    static get allowedOptions() { return false; }
```

Comment:
I thought we had decided that instead of returning false by default, this should return the list of commonOptions, i.e. the adapter gets only the common ones by default.